dffml/base.py

```diff
@@ -321,9 +321,17 @@ def wrapper(self, config: Optional[BaseConfig] = None, **kwargs):
                     )
                     raise
             if config is None:
-                raise TypeError(
-                    "__init__() missing 1 required positional argument: 'config'"
-                )
+                use_CONFIG = True
```

Comment:
Very close! We need to check that it `hasattr(self, "CONFIG")` before using `self.CONFIG`, this and the previous if statement should probably be combined.

So

```python
            elif config is None and hasattr(self, "CONFIG"):
                if kwargs:
                    # old logic
                else:
                    # your new logic
            else:
                raise TypeError(
                    "__init__() missing 1 required positional argument: 'config'"
                )
```

Reply:
Doesn't this also start throwing the error on other cases not handled previously because of the last `else` clause? Like `if config is not None:` while the kwargs condition may not be fulfilled, it will throw this error. Do we want that?